flexget/components/sites/sites/piratebay.py

```diff
@@ -130,76 +137,80 @@ def search(self, task, entry, config=None):
-                row = link.parent.parent.parent
-                description = row.find_all('a', attrs={'class': 'detDesc'})
-                if description and description[0].contents[0] == "piratebay ":
-                    logger.debug('Advertisement entry. Skipping.')
-                    continue
-                tds = row.find_all('td')
-                entry['torrent_seeds'] = int(tds[-2].contents[0])
-                entry['torrent_leeches'] = int(tds[-1].contents[0])
-                entry['torrent_availability'] = torrent_availability(
-                    entry['torrent_seeds'], entry['torrent_leeches']
-                )
-                # Parse content_size
-                size_text = link.find_next(attrs={'class': 'detDesc'}).get_text()
-                if size_text:
-                    size = re.search(r'Size (\d+(\.\d+)?\xa0(?:[PTGMK])?i?B)', size_text)
-                    if size:
-                        entry['content_size'] = parse_filesize(size.group(1))
-                    else:
-                        logger.error(
-                            'Malformed search result? Title: "{}", No size? {}',
-                            entry['title'],
-                            size_text,
-                        )
+            json_results = json.loads(page)
+            if len(json_results) == 0:
+                logger.error("Error while searching piratebay for %s." % search_string)
+            for result in json_results:
+                if result['id'] == '0':
+                    # JSON when no results found:
+                    # [{
+                    #    "id":"0",
+                    #    "name":"No results returned",
+                    #    "info_hash":"0000000000000000000000000000000000000000",
+                    #    "leechers":"0",
+                    #    "seeders":"0",
+                    #    "num_files":"0",
+                    #    "size":"0",
+                    #    "username":"",
+                    #    "added":"0",
+                    #    "status":"member",
+                    #    "category":"0",
+                    #    "imdb":"",
+                    #    "total_found":"1"
+                    # }]
+                    break
+                try:
+                    entry = Entry()
+                    entry['title'] = result['name']
+                    entry['torrent_seeds'] = int(result['seeders'])
+                    entry['torrent_leeches'] = int(result['leechers'])
+                    entry['torrent_timestamp'] = int(result['added'])
+                    entry['torrent_availability'] = torrent_availability(
+                        entry['torrent_seeds'], entry['torrent_leeches']
+                    )
+                    entry['content_size'] = result['size']
+                    entry['torrent_info_hash'] = result['info_hash']
+                    entry['url'] = UrlRewritePirateBay.info_hash_to_magnet(result['info_hash'], result['name'])
+                except KeyError:
```

Comment:
I don't see a reason for this. A KeyError when parsing a `result` seems unrecoverable to me